MantidQt/CustomInterfaces/src/EnggDiffraction/EnggDiffFittingPresenter.cpp

```diff
@@ -193,345 +207,475 @@ void EnggDiffFittingPresenter::fittingFinished() {
 // Fitting Tab Run Number & Bank handling here
 void EnggDiffFittingPresenter::fittingRunNoChanged() {
 
-  try {
-
-    // receive the run number from the text-field
-    auto strFocusedFile = m_view->getFittingRunNo();
+  // receive the run number from the text-field
+  const std::string userPathInput = m_view->getFittingRunNo();
 
-    if (m_previousInput == strFocusedFile) {
-      // Short circuit the checks and skip any warnings
-      // or errors as the user has not changed anything
-      // just clicked the box. Additionally this resolves an
-      // issue where QT will return the cursor and produce a new
-      // warning when the current warning is closed
-      return;
-    } else {
-      m_previousInput = strFocusedFile;
-    }
-
-    // file name
-    Poco::Path selectedfPath(strFocusedFile);
-    Poco::Path bankDir;
+  if (m_previousInput == userPathInput || userPathInput.empty()) {
+    // Short circuit the checks and skip any warnings
+    // or errors as the user has not changed anything
+    // just clicked the box. Additionally this resolves an
+    // issue where QT will return the cursor and produce a new
+    // warning when the current warning is closed
+    return;
+  } else {
+    m_previousInput = userPathInput;
+  }
 
-    std::vector<std::string> runnoDirVector;
+  // file name
+  const Poco::Path pocoUserPathInput(userPathInput);
 
-    std::string strFPath = selectedfPath.toString();
-    // returns empty if no directory is found
+  std::vector<std::string> foundFullFilePaths;
 
-    // split directory if 'ENGINX_' found by '_.'
-    std::vector<std::string> splitBaseName;
-    if (strFPath.find("ENGINX_") != std::string::npos) {
-      boost::split(splitBaseName, strFPath, boost::is_any_of("_."));
-    }
+  // returns empty if no directory is found
+  const std::string parsedUserInput = pocoUserPathInput.toString();
 
-    // runNo when single focused file selected
-    std::vector<std::string> runNoVec;
+  // split directory if 'ENGINX_' found by '_.'
+  std::vector<std::string> splitBaseName;
+  if (parsedUserInput.find("ENGINX_") != std::string::npos) {
```

Comment:
Beware of the hardcoded "ENGINX". We'd eventually like to get rid of them. What is needed here is `currentInstrument()` which at the moment is part of `IEnggDiffractionView` and therefore not accesible from the fitting presenter. I think that this method and `getRBNumber()` should be moved to a different interface (probably an interface of its own) similarly as `IEnggDiffractionUserMsg` or `IEnggDiffractionSettings`. But that's not the subject of this PR. It could be a lower priority maintenance issue/task.